src/ops/conv2d_test.ts

```diff
@@ -74,6 +74,33 @@ describeWithFlags('conv2d', ALL_ENVS, () => {
     expectArraysClose(result, [2, 4, 6, 8]);
   });
 
+  it('packed x=[2,1,2] f=[1,1,2,2] s=1 d=1 p=0', () => {
+    const inputShape: [number, number, number] = [2, 1, 2];
+    const fSize = 1;
+    const pad = 0;
+    const stride = 1;
+
+    const x = tf.tensor3d([1, 2, 3, 4], inputShape);
+    const w = tf.tensor4d([1, 2, 3, 4], [fSize, fSize, 2, 2]);
+
+    const webglLazilyUnpackFlagSaved = tf.ENV.get('WEBGL_LAZILY_UNPACK');
+    tf.ENV.set('WEBGL_LAZILY_UNPACK', true);
+    const webglLazilyPackBinaryOperationsFlagSaved =
+        tf.ENV.get('WEBGL_PACK_BINARY_OPERATIONS');
+    tf.ENV.set('WEBGL_PACK_BINARY_OPERATIONS', true);
+    
+    // |result| is packed for WebGL backends.
+    const result = tf.conv2d(x, w, stride, pad);
+    const result1 = tf.conv2d(result, w, stride, pad);
```

Comment:
Can you help me understand the role of this test? Why does applying convolution twice uniquely test your new kernel? There should be other tests in our conv2d suite that hit your kernel, right? Please include your explanation in a comment for this test.

Reply:
Changed the comment:

```
    // First conv2D tests conv2D with non-packed input |x|, and the second uses
    // packed input |result|.
    const result = tf.conv2d(x, w, stride, pad);
    const result1 = tf.conv2d(result, w, stride, pad);

```

MathBackendWebGL.conv2dByMatMul optimizes the case when input is packed.
All other cases hitting the test have non-packed input.
That said, I could replace the first "`const result = tf.conv2d(x, w, stride, pad);`" by any other operation producing packed output.